### PR TITLE
Delete Note Bug Fixed on Reviewer

### LINKED ISSUE #14460: [BUG][Crash][2.17alpha]: "Delete note" on the reviewer causes crash
### Checked for duplicates?

- [X] This issue is not a duplicate

### What are the steps to reproduce this bug?

1. Use the HEAD (ad22652) or a recent one
2. Open the app
3. Tap a deck
4. "Delete note" on the reviewer
5. Select "Delete" in the confirmation dialog box

### Expected behaviour

No crash

### Actual behaviour

Crash

### Debug info

```text
AnkiDroid Version = 2.17alpha1-debug (ad2265285a3d8fc45dc444a6653963fab388aad1)

Android Version = 11

ProductFlavor = amazon

Manufacturer = OPPO

Model = CPH2013

Hardware = qcom

Webview User Agent = Mozilla/5.0 (Linux; Android 11; CPH2013 Build/RKQ1.200903.002; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/116.0.0.0 Mobile Safari/537.36

ACRA UUID = f74d8c3b-b173-4337-ae0c-819d3934bd5e

Crash Reports Enabled = false

DatabaseV2 Enabled = true
```

### (Optional) Anything else you want to share?

- At the card browser, "Delete notes" doesn't cause crash. 
- Not reproduced on 2.16.5 and the released 2.17alpha1

### Research

- [X] I am reporting a bug specific to AnkiDroid (Android app)
- [X] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] (Optional) I have confirmed the issue **_has not yet occurred_** in the latest alpha release  ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### PR DESCRIPTION
**Pull request by soCallmeAdityaKumar**
## Purpose / Description
"Delete note" on the reviewer causes crash 

## Fixes
#14460

## Approach
Added a Snackbar to show the note deleted

## Checklist

- [✔️] You have a descriptive commit message with a short title (first line, max 50 chars).
- [ ] You have commented your code, particularly in hard-to-understand areas
- [✔️] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.kt [hunk 1]
```diff
@@ -830,10 +831,15 @@ abstract class AbstractFlashcardViewer :
     private fun deleteNoteWithoutConfirmation() {
         val cardId = currentCard!!.id
         launchCatchingTask {
-            withProgress() {
+            val noteCount = withProgress("Deleting the note") {
```
Comment (dae) [inline AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.kt:834]: I suggest removing the text to use a generic 'processing' instead, as this text is not translatable.

Comment (soCallmeAdityaKumar) [reply AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.kt]: Hey @dae I have removed the text, do i need to do something else.

File: AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.kt [hunk 2]
```diff
@@ -127,6 +127,7 @@ abstract class AbstractFlashcardViewer :
     private var mReplayOnTtsInit = false
     private var mAnkiDroidJsAPI: AnkiDroidJsAPI? = null
     var server: ReviewerServer? = null
+    private var mUndoSnackbar: Snackbar? = null
```
Comment (dae) [inline AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.kt:130]: - In new Kotlin code, please avoid the m prefix.
- This doesn't currently appear to be used - what future usage did you have in mind?

**Comment by criticalAY** [comment]
i already suggested to remove this var as it is not being used anywhere

**Comment by dae** [comment]
Do you mean you suggested it on Discord? Any such suggestions are best placed here :-)

**Comment by criticalAY** [comment]
No no when I reviewed this PR i dropped a comment that its not used anywhere and that snackbar can be used directly

**Comment by dae** [comment]
I don't see a review by you - are you sure you submitted it?

**Comment by criticalAY** [comment]
umm this is interesting I did one and I am able to see it, maybe some error
image

**Review by criticalAY — APPROVED** [review]
So far looks good to me

File: AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.kt [hunk 3]
```diff
@@ -830,10 +831,15 @@ abstract class AbstractFlashcardViewer :
...
+            val noteCount = withProgress() {
                 undoableOp {
                     removeNotes(cids = listOf(cardId))
-                }
+                }.count
+            }
+            val deletedMessage = resources.getQuantityString(R.plurals.card_browser_cards_deleted, noteCount, noteCount)
+            showSnackbar(deletedMessage, Snackbar.LENGTH_LONG) {
+                setAction(R.string.undo) { launchCatchingTask { undoAndShowSnackbar() } }
+                mUndoSnackbar = this
```
[the remark this answers is not in the document]
Comment (soCallmeAdityaKumar) [reply AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.kt]: i removed the mSnackbar

**Comment by dae** [comment]
Looks good - thanks! FYI, the issue here was rather surprising, and has been reported on https://youtrack.jetbrains.com/issue/KT-62158/Outer-scope-coerces-inner-block-to-Unit

**Comment by snowtimeglass** [comment]
Thank you for the fix!

**Pull request merged**